### PR TITLE
docs: updated verious util function docs, added tests, refactoring

### PR DESCRIPTION
**Pull request by dsperac**
## Motivation and Resolution

...

### RPC version (if applicable)

...

## Usage related changes

- Change 1.
- ...

## Development related changes

- Change 1.
- ...

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: src/utils/contract.ts [hunk 1]
```diff
@@ -30,10 +34,12 @@ export function isSierra(
...
  *
  * @param {DeclareContractPayload} payload - The payload containing contract information.
- *
  * @return {CompleteDeclareContractPayload} - The `CompleteDeclareContractPayload` with extracted contract hashes.
- *
  * @throws {Error} - If extraction of compiledClassHash or classHash fails.
+ * @example
+ * ```typescript
+ * const result = extractContractHashes(contract);
+ * ```
```
Comment (ivpavici) [inline src/utils/contract.ts:42]: missing what is the result ?

File: src/utils/contract.ts [hunk 2]
```diff
@@ -18,6 +18,10 @@ import { isString } from './shortString';
  *
  * @param {CairoContract | string} contract - The contract to check. Can be either a CairoContract object or a string representation of the contract.
  * @return {boolean} - Returns true if the contract is a Sierra contract, otherwise false.
+ * @example
+ * ```typescript
+ * const result = isSierra(contract);
+ * ```
```
Comment (ivpavici) [inline src/utils/contract.ts:24]: missing what is the result ?

File: src/utils/json.ts [hunk 3]
```diff
@@ -2,35 +2,69 @@ import * as json from 'lossless-json';
 
 /**
  * Convert string to number or bigint based on size
+ * @param str string to parse
+ * @return {number | BigInt} parsed number
+ * @example
+ * ```typescript
+ * parseIntAsNumberOrBigInt('1.2'); // 1.2
+ * parseIntAsNumberOrBigInt('11223344556677889900'); // 11223344556677890048n
+ * ```
```
Comment (ivpavici) [inline src/utils/json.ts:11]: missing 
// result ...

File: src/utils/events.ts [hunk 4]
```diff
@@ -6,7 +6,23 @@ import { cleanHex } from './num';
...
+ *     encodeShortString('Token'),
+ *     encodeShortString('ERC20'),
+ *     account.address,
+ *   ],
+ *   salt,
+ *   unique: true,
+ * });
+ * const txReceipt = await provider.waitForTransaction(deployment.transaction_hash);
+ * const udcEvent = parseUDCEvent(txReceipt as any);
+ * ```
```
Comment (ivpavici) [inline src/utils/events.ts:25]: missing
// result ...

File: src/utils/json.ts [hunk 5]
```diff
@@ -1,36 +1,63 @@
...
+ * @param value JSON object
+ * @param [replacer] Function that alters the behavior of the stringification process
+ * @param [space] Used to insert white space into the output JSON string
+ * @param [numberStringifiers] Function used to stringify numbers (returning undefined will delete the property from the object)
+ * @return {string} JSON string
+ * @example
+ * ```typescript
+ * const str = '[123, 12.3, 11223344556677889900]';
+ * const result = stringify(str);
+ * // result = [123, 12.3, 11223344556677890048n]
```
Comment (penovicp) [inline src/utils/json.ts:59]:
```suggestion
 * const value = [123n, 12.3, 11223344556677889900n];
 * const result = stringify(value);
 * // result = '[123,12.3,11223344556677890048]'
```

**Review by penovicp — APPROVED** [review]
Noticed one input/output swap in an example comment, otherwise lgtm

**Review by penovicp — APPROVED** [review]

**Pull request merged**